lib/defender/aws_defender.c

```diff
@@ -276,11 +281,42 @@ static DefenderState_t prvStateSubscribe( void )
     return eDefenderStateSubscribeMqttSuccess;
 }
 
+static char * prvGetDefenderMqttTopicString( char * pcTopicTemplate )
+{
+    char * pcTopic = NULL;
+    uint32_t ulTopicLength = 0;
+    uint32_t ulCharsWritten = 0;
+
+    /* Determine the maximum resulting string length. */
+    ulTopicLength = 1 + strlen( pcTopicTemplate ) + strlen( clientcredentialIOT_THING_NAME );
```

Comment:
This should subtract off the characters for the '%s' included in the formatting string... otherwise, when it goes to zero out the last character further down, there could be a couple extra chars at the end of the topic length that are unintended.

Reply:
Actually snprintf will put the null terminator in the correct place... so we can both drop the extra 2 characters and remove the part zeroing out the last character